tools/psa/tfm/bin_utils/imgtool.py

```diff
@@ -0,0 +1,177 @@
+#! /usr/bin/env python3
+#
+# Copyright 2017 Linaro Limited
+# Copyright (c) 2018, Arm Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import os
+import re
+import argparse
+from imgtool_lib import keys
+from imgtool_lib import image
+from imgtool_lib import version
+import sys
+
+def find_load_address(args):
+    load_address_re = re.compile(r"^#define\sIMAGE_LOAD_ADDRESS\s+(0x[0-9a-fA-F]+)")
+
+    if os.path.isabs(args.layout):
+            configFile = args.layout
+    else:
+        scriptsDir = os.path.dirname(os.path.abspath(__file__))
+        configFile = os.path.join(scriptsDir, args.layout)
+
+    ramLoadAddress = None
+    with open(configFile, 'r') as flash_layout_file:
+        for line in flash_layout_file:
+            m = load_address_re.match(line)
+            if m is not None:
+                ramLoadAddress = int(m.group(1), 0)
+                print("**[INFO]** Writing load address from the macro in "
+                      "flash_layout.h to the image header.. "
```

Comment:
In the future, if this script is going to be reused by other targets, we should be moving "configuration" into the `targets.json`/`mbed_lib.json`, not scraping it from the `flash_layout.h` file.

Reply:
I absolutely agree, this is something we are aware of and will try to use this in the Musca PR

Reply:
@mikisch81 @orenc17 Can y'all make and link a Jira issue so that we don't lose this conversation?

We'll start CI in the meanwhile.

Reply:
@cmonr @bridadan @orenc17 @alzix Created [IOTSEC-1104](https://jira.arm.com/browse/IOTSEC-1104) for that.